### PR TITLE
Simple multiplayer, just moving player on other person's screen

### PR DESCRIPTION
**Pull request by jcr7467**
Added multiplayer file, split up classes into a new file and imported them. added socket functionality so that the client can see their opponents position. Added functions to server to simplify emitting inside of socket connection

### CODE REVIEW

**Comment by jcr7467** [comment]
@all, just look through it to at least be somewhat familiar with it @asrendon24 couldn't add you as reviewer, take a look

**Review by antonalcala — APPROVED** [review]
yupp let's go

File: server/server.js [hunk 1]
```diff
@@ -84,25 +84,63 @@ let sessionData = session({
...
+    socket.on('updateMyCoordinates', request => {
+        const {roomID} = request;
+        const {xCoord} = request;
+        const {yCoord} = request;
+        socket.broadcast.to(roomID).emit('updateOpponentCoordinates', {
+            xCoord: xCoord,
+            yCoord: yCoord
+        })
+    });
+};
```
Comment (randallwc) [inline server/server.js:105]: need to add functionality for multiple players in one room

File: server/server.js [hunk 2]
```diff
@@ -84,25 +84,63 @@ let sessionData = session({
...
 
+const listenForLaser = (socket) => {
+    socket.on('laserShot', request => {
+        const {roomID} = request;
+        if (!!roomID){
+            socket.broadcast.to(roomID).emit('opponentLasered')
```
Comment (randallwc) [inline server/server.js:111]: multiple players

File: server/server.js [hunk 3]
```diff
@@ -84,25 +84,63 @@ let sessionData = session({
...
-    socket.emit("WelcomeClient", {
-        message: "Welcome to sky danger ranger! we're glad to have you here. It's gonna be a ride!"
-    })
-
-    // Client sends message
-    socket.on('clientMessageToServer', request => {
-        console.log(`Message from Client with id ${socket.id}: ${request.message}`);
+        // socket.broadcast.to: send to all in room except person who sent it
+        socket.broadcast.to(roomID).emit("newRoomMemberJoined", {
+            message: "New member has joined!"
```
Comment (randallwc) [inline server/server.js:125]: broadcast unique id

File: main_multiplayer.py [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+        if event.type == pygame.QUIT:
+            sys.exit()
+        # check for mouse clicks
+        if event.type == pygame.MOUSEBUTTONDOWN:
+            isClicking = True
+        if event.type == pygame.MOUSEBUTTONUP:
+            isClicking = False
+            lineWidth = maxLineWidth
+
+    pygame.display.update()
```
Comment (randallwc) [inline main_multiplayer.py:144]: main loop should be in it's own file

File: main_multiplayer.py [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+
+@sock.on('messageFromRoomMember')
+def message_from_room_member(data):
+    print("new message from a room member!")
+    print(data)
+
+@sock.on('updateOpponentCoordinates')
+def update_opponent_coordinates(data):
+    print(data['xCoord'], data['yCoord'])
+    OpponentRanger.UpdateCoords(data['xCoord'], data['yCoord'])
```
Comment (randallwc) [inline main_multiplayer.py:52]: should be in it's own file separate from main loop

File: common/gameobjects.py [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+- [x] make plane follow cursor
+- [ ] make multiple clouds appear
+- [ ] make enemies appear
+- [ ] make enemies take damage
+- [ ] make enemies deal damage
+- [ ] make amo boxes appear
+- [ ] make plane shoot
+- [ ] make plane GIF render plane flying
+- [ ] ...
+'''
```
Comment (randallwc) [inline common/gameobjects.py:24]: should be in main file

**Review by randallwc — CHANGES_REQUESTED** [review]
- needs to support multiple connections before the merge
- needs to have 3 files (main loop, single-player, multiplayer)

**Pull request closed without merging**